logstash-core/lib/logstash/agent.rb

```diff
@@ -243,6 +243,19 @@ def running_pipelines?
     end
   end
 
+  def running_user_defined_pipelines?
+    user_defined_pipelines.any?
+  end
+
+  def user_defined_pipelines
+    @pipelines_mutex.synchronize do
+      @pipelines.select do |pipeline_id, _|
+        pipeline = @pipelines[pipeline_id]
+        pipeline.running? && !pipeline.system?
```

Comment:
since this method's name is `user_defined_pipelines` it may not make sense to filter out pipelines that aren't running.